parlai/core/torch_generator_agent.py

```diff
@@ -1047,6 +1066,24 @@ def _treesearch_factory(self, device, verbose=False):
                 verbose=verbose,
                 gpu_beam_blocking=self.opt.get('gpu_beam_blocking', False),
             )
+        elif method == 'factual_nucleus':
+            return FactualNucleusSampling(
+                self.opt['topp'],
+                self.opt['lambda_decay'],
+                self.opt['omega_bound'],
+                self.opt['p_reset'],
+                beam_size,
+                min_length=self.beam_min_length,
+                block_ngram=self.beam_block_ngram,
+                context_block_ngram=self.beam_context_block_ngram,
+                length_penalty=self.opt.get('beam_length_penalty', 0.65),
+                padding_token=self.NULL_IDX,
+                bos_token=self.START_IDX,
+                eos_token=self.END_IDX,
+                device=device,
+                verbose=verbose,
+                dict=self.dict,
+            )
```

Comment:
is `gpu_beam_blocking` needed as in other sampling methods?

Reply:
yes, will add thanks